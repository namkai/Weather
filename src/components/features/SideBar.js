import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import * as actions from '../../actions/index';
import citySelector from '../../selectors/selected_stored_cities';
import Cities from '../common/Cities/index';
import SearchBar from '../common/SearchBar';
import SideBarLayout from '../layouts/SideBarLayout';


export const SideBar = ({ storedCities, fetchWeather, updateStoredCities, clearStoredCities }) => (
  <SideBarLayout searchBar={<SearchBar fetchWeather={fetchWeather}/>}>
    <Cities
      storedCities={storedCities}
      fetchWeather={fetchWeather}
      updateStoredCities={updateStoredCities}
      clearStoredCities={clearStoredCities} />
  </SideBarLayout>
);

SideBar.defaultProps = {
  storedCities: PropTypes.arrayOf(PropTypes.string, PropTypes.object),
  fetchWeather: PropTypes.func,
  updateStoredCities: PropTypes.func,
  clearStoredCities: PropTypes.func,
};

SideBar.propTypes = {
  storedCities: PropTypes.arrayOf(PropTypes.string, PropTypes.object),
  fetchWeather: PropTypes.func,
  updateStoredCities: PropTypes.func,
  clearStoredCities: PropTypes.func,
};

export default connect(state => ({ storedCities: citySelector(state) }), actions)(SideBar);
