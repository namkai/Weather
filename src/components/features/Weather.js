import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import CurrentWeather from '../common/CurrentWeather';
import WeatherHeader from '../common/WeatherHeader';
import selectedForecast from '../../selectors/selected_forecast';
import Chart from '../common/Chart';

export const Weather = ({ weather, forecast }) => (
  <div className="col-md-9 content">
    <WeatherHeader city={weather.name} description={weather.weather}/>
    <hr className="mt-3"/>
    <CurrentWeather {...weather} />
    <Chart forecast={forecast} />
  </div>
);

Weather.defaultProps = {
  weather: {},
  forecast: {
    list: [],
  },
};

Weather.propTypes = {
  weather: PropTypes.object,
  forecast: PropTypes.arrayOf(PropTypes.array),
};

const mapStateToProps = state => ({
  weather: state.weather,
  forecast: selectedForecast(state),
});


export default connect(mapStateToProps)(Weather);
