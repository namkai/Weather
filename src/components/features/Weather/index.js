import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import CurrentWeather from '../../common/CurrentWeather/index';
import WeatherHeader from '../../common/WeatherHeader/view';
import Chart from '../../common/Chart/index';
import WeatherLayout from '../../layouts/WeatherLayout';

export const Weather = ({ weather, forecast }) => (
  <WeatherLayout header={<WeatherHeader city={weather.name} description={weather.weather} />} >
    <CurrentWeather {...weather} />
    <Chart forecast={forecast} />
  </WeatherLayout>
);

Weather.defaultProps = {
  weather: {},
  forecast: {
    list: [],
  },
};

Weather.propTypes = {
  weather: PropTypes.object,
  forecast: PropTypes.arrayOf(PropTypes.array),
};

export default Weather;
