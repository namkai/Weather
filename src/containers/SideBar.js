import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import * as actions from '../actions/index';
import Cities from '../components/Cities';
import SearchBar from '../components/SearchBar';
import citySelector from '../selectors/selected_stored_cities';

export const SideBar = ({ storedCities, fetchWeather, updateQuery, input, updateStoredCities }) => (
  <div className="col-md-3 sidebar">
    <nav className="sidebar-nav">
      <div className="sidebar-header">
        <button
          className="nav-toggler nav-toggler-md sidebar-toggler" type="button" data-toggle="collapse"
          data-target="#nav-toggleable-md"
        >
          <span className="sr-only">Toggle nav</span>
        </button>
        <a className="sidebar-brand img-responsive" href="#">
          <span style={{ color: '#1ec8ff' }} className="icon icon-drop sidebar-brand-icon" />
        </a>
      </div>

      <div className="collapse nav-toggleable-md" id="nav-toggleable-md">
        <SearchBar updateQuery={updateQuery} fetchWeather={fetchWeather} input={input} />
        <Cities
          storedCities={storedCities}
          fetchWeather={fetchWeather}
          updateStoredCities={updateStoredCities}
        />
        <hr className="visible-xs mt-3" />
      </div>
    </nav>
  </div>
	);

SideBar.defaultProps = {
	storedCities: PropTypes.arrayOf(PropTypes.object),
	fetchWeather: PropTypes.func,
	updateQuery: PropTypes.func,
	input: PropTypes.string,
};

SideBar.propTypes = {
	storedCities: PropTypes.arrayOf(PropTypes.object),
	fetchWeather: PropTypes.func,
	updateQuery: PropTypes.func,
	input: PropTypes.string,
};

const mapStateToProps = state => ({
	storedCities: citySelector(state),
	input: state.input,
});

export default connect(mapStateToProps, actions)(SideBar);
