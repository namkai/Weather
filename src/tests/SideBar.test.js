import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { SideBar } from '../components/features/SideBar';

it('renders correctly', () => {
	const sidebar = shallow(<SideBar />);
	expect(toJson(sidebar)).toMatchSnapshot();
});
