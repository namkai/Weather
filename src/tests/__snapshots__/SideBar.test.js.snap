// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="col-md-3 sidebar"
>
  <nav
    className="sidebar-nav"
  >
    <div
      className="sidebar-header"
    >
      <button
        className="nav-toggler nav-toggler-md sidebar-toggler"
        data-target="#nav-toggleable-md"
        data-toggle="collapse"
        type="button"
      >
        <span
          className="sr-only"
        >
          Toggle nav
        </span>
      </button>
      <a
        className="sidebar-brand img-responsive"
        href="#"
      >
        <span
          className="icon icon-drop sidebar-brand-icon"
          style={
            Object {
              "color": "#1ec8ff",
            }
          }
        />
      </a>
    </div>
    <div
      className="collapse nav-toggleable-md"
      id="nav-toggleable-md"
    >
      <SearchBar
        fetchWeather={[Function]}
        input=""
        updateQuery={[Function]}
      />
      <Cities
        fetchWeather={[Function]}
        storedCities={[Function]}
      />
      <hr
        className="visible-xs mt-3"
      />
    </div>
  </nav>
</div>
`;
